example/poisson/poisson_package.cpp

```diff
@@ -49,59 +50,123 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
 
   auto mrho = Metadata({Metadata::Cell, Metadata::Derived, Metadata::OneCopy});
   pkg->AddField("density", mrho);
+  pkg->AddField("rhs", mrho);
 
   auto mphi = Metadata({Metadata::Cell, Metadata::Independent, Metadata::FillGhost});
   pkg->AddField("potential", mphi);
 
+  int ndim = 1 + (pin->GetInteger("parthenon/mesh", "nx2") > 1) +
+             (pin->GetInteger("parthenon/mesh", "nx3") > 1);
+  // set up the stencil object corresponding to the finite difference
+  // discretization we adopt in this pacakge
+  const int nstencil = 1 + 2 * ndim;
+  const Real w0 = 1.0 / (2.0 * ndim);
+  std::vector<Real> wgts({w0, -1.0, w0, w0, w0, w0, w0});
+  std::vector<std::vector<int>> offsets(
+      {{-1, 0, 1, 0, 0, 0, 0}, {0, 0, 0, -1, 1, 0, 0}, {0, 0, 0, 0, 0, -1, 1}});
+
+  auto stencil = parthenon::solvers::Stencil<Real>("stencil", nstencil, wgts, offsets);
+  pkg->AddParam<>("stencil", stencil);
+
   return pkg;
 }
 
 auto &GetCoords(std::shared_ptr<MeshBlock> &pmb) { return pmb->coords; }
 auto &GetCoords(Mesh *pm) { return pm->block_list[0]->coords; }
 
 template <typename T>
-TaskStatus UpdatePhi(T *u, T *du) {
-  Kokkos::Profiling::pushRegion("Task_Poisson_UpdatePhi");
+TaskStatus SumMass(T *u, Real *reduce_sum) {
   auto pm = u->GetParentPointer();
 
   IndexRange ib = u->GetBoundsI(IndexDomain::interior);
   IndexRange jb = u->GetBoundsJ(IndexDomain::interior);
   IndexRange kb = u->GetBoundsK(IndexDomain::interior);
 
   PackIndexMap imap;
-  const std::vector<std::string> vars({"density", "potential"});
+  const std::vector<std::string> vars({"density"});
   const auto &v = u->PackVariables(vars, imap);
   const int irho = imap["density"].first;
-  const int iphi = imap["potential"].first;
-  const std::vector<std::string> phi_var({"potential"});
-  PackIndexMap imap2;
-  const auto &dv = du->PackVariables(phi_var, imap2);
-  const int idphi = imap2["potential"].first;
 
   auto coords = GetCoords(pm);
   const int ndim = v.GetNdim();
   const Real dx = coords.Dx(X1DIR);
   for (int i = X2DIR; i <= ndim; i++) {
     const Real dy = coords.Dx(i);
     PARTHENON_REQUIRE_THROWS(dx == dy,
-                             "UpdatePhi requires that DX be equal in all directions.");
+                             "ComputeRHS requires that DX be equal in all directions.");
+  }
+
+  Real total;
+  parthenon::par_reduce(
+      parthenon::loop_pattern_mdrange_tag, "ComputeRHS", DevExecSpace(), 0,
```

Comment:
Loop name should be `SumMass` or something to avoid ambiguity with the `ComputeRHS` task

Reply:
oops copy/paste error.  got it.